actionview/lib/action_view/helpers/form_helper.rb

```diff
@@ -2479,6 +2520,25 @@ def emitted_hidden_id? # :nodoc:
       end
 
       private
+        def tag_id(method_name, index = nil)
```

Comment:
This set of changes was lifted directly from `ActionView::Helpers::Tags::Base`.

https://github.com/rails/rails/blob/7101489293ece071013417aeed2da3aa1b0927c9/actionview/lib/action_view/helpers/tags/base.rb#L119-L137

Given the "a little duplication to construct fewer strings", it seemed like that code was finely tuned for performance, which made me hesitant to abstract out a shared and re-usable chunk of code.

My gut tells me that `FormBuilder#id` will be invoked less frequently than the `Tags::Base` code. That being said, if there _is_ an abstraction that feels right, I'm all for extracting it.